client/components/generate-date.jsx

```diff
@@ -18,9 +20,14 @@ export default class GenerateDate extends React.Component {
   }
 
   componentDidMount() {
+    const { token } = this.context.token;
+    this.setState({ token });
```

Comment:
You can use `this.context.token` and remove it from your state.

Reply:
Corrected